flink-runtime/src/main/java/org/apache/flink/runtime/operators/Driver.java

```diff
@@ -87,4 +87,6 @@
 	 * @throws Exception Exceptions may be forwarded.
 	 */
 	void cancel() throws Exception;
+
+	void resetForIterativeTasks() throws Exception;
```

Comment:
I think you should use the already existing `ResettableDriver` interface instead of introducing a new resetting method.